source/StoryMenuState.hx

```diff
@@ -396,4 +410,27 @@ class StoryMenuState extends MusicBeatState
 		intendedScore = Highscore.getWeekScore(curWeek, curDifficulty);
 		#end
 	}
+
+	public static function unlockNextWeek(week:Int):Void
+	{
+		// TODO: get the weekData from class (but making it static makes it dissapear when beating the week)
```

Comment:
if your pr isn't ready you're supposed to make it a draft pr not a ready-for-review/open pr

Reply:
yeah I wanted to make that a todo, that I can't think of a fix rn

Reply:
would you have any idea as to why that happens?

Reply:
well to clarify when for example you beat week 1, all strings in weekData[1] get nulled